src/stories/Library/material-search/material-search.scss

```diff
@@ -0,0 +1,235 @@
+  margin-bottom: $_gin_xs_spacing;
+  font-weight: $_gin_font_weight_semibold;
+}
+
+.material-search__inputs-container {
+  display: grid;
+  grid-template-columns: 2fr minmax(auto, 220px);
+  gap: $_gin_m_spacing;
+  margin-bottom: $_gin_xs_spacing;
+}
+
+.material-search__inputs-container--hidden {
+  display: none; // This class is only applied in Drupal.
+}
+
+// Preview
+.material-search__preview {
+  color: $color__global-grey;
+  cursor: not-allowed;
+  display: grid;
+  background-color: $color__global-white;
+  min-height: 170px;
+  align-items: center;
+  border: 1px solid $_gin_border_color;
+  border-radius: $_gin_xs_spacing;
+  overflow: hidden;
+}
+
+.material-search__preview-material {
+  display: grid;
+  padding: 0 $_gin_m_spacing;
+  grid-template-columns: min-content auto;
+  column-gap: $_gin_m_spacing;
+  align-items: center;
+  height: 100%;
+
+  .cover {
+    height: $MATERIAL_SMALL_MOBILE;
+    @media screen and (min-width: 768px) {
+      height: $MATERIAL_SMALL_DESKTOP;
+    }
+  }
+}
+
+.material-search__preview-empty {
+  height: 100%;
+  display: flex;
+  justify-content: center;
+  background: $color__global-tertiary-1;
+  align-items: center;
+  color: $color__global-grey;
+}
+
+.material-search__preview-loading {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100%;
+  flex-direction: column;
+}
+
+.material-search__input,
+.material-search__selector {
+  border: 1px solid $_gin_border_color;
+  padding: $_gin_xs_spacing;
+  border-radius: $_gin_xs_spacing;
+  height: 100%;
+  font-family: inherit;
+
+  &:disabled {
+    background-color: $color__global-tertiary-1;
+  }
+}
+
+.material-search__preview-content {
+  display: grid;
+  grid-template-columns: max-content auto;
+  column-gap: $_gin_m_spacing;
+}
+
+.material-search__preview-content__label {
+  font-weight: 700;
+}
+
+// Loading
+.material-search__loading {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+}
+
+.material-search__loading-spinner {
+  width: $_loading-spinner-size;
+  height: $_loading-spinner-size;
+  animation: spin 0.9s linear infinite;
+  animation-timing-function: ease-in-out;
+}
+
+.material-search__loading-text {
+  font-size: 1rem;
+}
+
+@keyframes spin {
+  0% {
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+}
+
+// Search List
+.material-search__list {
+  border: 1px solid $_gin_border_color;
+  margin-top: $_gin_m_spacing;
+  overflow: hidden;
+  border-radius: $_gin_xs_spacing;
+  background-color: $color__global-white;
+  position: relative;
+  overflow-y: auto;
+  height: 100%;
+  max-height: 500px;
+}
+
+.material-search__list-results {
+  display: flex;
+  flex-direction: column;
+  justify-content: left;
+  row-gap: $_gin_xs_spacing;
+  padding: $_gin_xs_spacing;
+  padding-bottom: 20px;
+  background-color: $_color__background-light;
+  margin: 0;
+}
+
+.material-search__list-result-button {
```

Comment:
Your BEM structure relies on putting everything in one block, `material-search` and then adding elements where the name implies more and more nesting, `list`, `list-results`, `list-result-button`. It is OK for you to break the element into more blocks.